### ISSUE TITLE
BrowserSync client script not cached

### ISSUE THREAD

**Issue by maoueh** [BrowserSync/browser-sync#657]
Hi there,

It seems that the browser-sync client javascript that is served to the browser has no caching headers in the HTTP response leading the browser to not cache it upon new requests to the file.

I'm wondering why because other files served through browser sync correctly have the caching headers (specially the `Cache-Control` and `Etag` ones). I am missing something obvious here?

Here the steps to replicate (I could create a small repo if you prefer):

    $ mkdir /tmp/browser-sync-client && cd /tmp/browser-sync-client
    $ echo "<html>Browser Sync Caching Test</html>" > index.html
    $ browser-sync start --server &
    $ # You can also check in browser developer tools
    $ curl -v http://localhost:3000 2>&1 | grep "Cache\|ETag" # Correct, has caching headers
    $ curl -v http://localhost:3000/browser-sync/browser-sync-client.2.7.6.js 2>&1 | grep "Cache-Control\|ETag" # Incorrect, no caching headers

I'm willing to tackle this issue if you aim me in the right direction.

Regards,
Matt

**Comment by shakyShane** [comment]
@maoueh we'd love to receive a contribution from you!

so it's actually an issue in the `browser-sync-client` library...

Browsersync [retrieves](https://github.com/BrowserSync/browser-sync/blob/master/lib/async.js#L170-L174) the middleware via a plugin manager and then [registers it](https://github.com/BrowserSync/browser-sync/blob/master/lib/server/static-server.js#L26-L30) with the connect middleware stack. (it does the same thing for the proxy/snippet servers too)

so basically, you can add headers to the client library [here](https://github.com/BrowserSync/browser-sync-client/blob/master/index.js#L23-L26)

Hope that helps

**Comment by maoueh** [comment]
@shakyShane Excellent, thanks for the information, that's all I needed. I will take a look at it this week. Will ping you back if I need further help.

Regards,
Matt

**Comment by maoueh** [comment]
@shakyShane Let's move discussion to the [WIP PR](https://github.com/BrowserSync/browser-sync-client/pull/19). You're early comments are welcome.
